bottlerocket/types/src/agent_config.rs

```diff
@@ -92,12 +92,15 @@ impl Configuration for MigrationConfig {}
 #[serde(rename_all = "camelCase")]
 pub struct EksClusterConfig {
     /// The name of the eks cluster to create or an existing cluster.
+    /// If an eksctl config is passed, the names must match.
```

Comment:
Below, the docstring for `encoded_eksctl_config` says 

>This conflicts with `cluster_name`, `zones` and `version`. If an encoded cluster config is provided the other values will not be used.

So this wouldn't be accurate right? Can we turn `cluster_name` into an option? I'd rather have `encoded_eksctl_config` be the source of truth if it's specified.

Reply:
The only way to do that would be to serialize the encoded config. Since we use eksctl and the aws sdk we have to know the cluster name.

Reply:
I don't think it'd be that much overhead to deserialize the encoded cluster config and extract out the `cluster_name`, `region` If they don't exist, they should, then we fallback to defaults or error. Like I mentioned in another comment, we're already using `serde_yaml` elsewhere.